src/wh_nvm_flash.c

```diff
@@ -86,6 +91,10 @@ static int nfPartition_ProgramStart(whNvmFlashContext* context, int partition,
 static int nfPartition_ProgramCount(whNvmFlashContext* context, int partition,
         uint32_t count);
 static int nfPartition_ProgramInit(whNvmFlashContext* context, int partition);
+static int nfPartition_CheckRangeUnits(whNvmFlashContext* context,
+                                       int partition,
+                                       uint32_t units_offset,
+                                       uint32_t units_count);
```

Comment:
Changed to match byte-based solution below.
```suggestion
static int nfPartition_CheckDataRange(whNvmFlashContext* context,
                                       int partition,
                                       uint32_t byte_offset,
                                       uint32_t byte_count);
```